src/main/java/reposense/git/GitConfig.java

```diff
@@ -0,0 +1,84 @@
+import reposense.system.LogsManager;
+
+/**
+ * Contains git config related functionalities.
+ * Git config is used to set Git configuration values on a global or local
+ * project level.
+ */
+public class GitConfig {
+    public static final String FILTER_LFS_SMUDGE_KEY = "filter.lfs.smudge";
+    public static final String FILTER_LFS_SMUDGE_VALUE = "git-lfs smudge --skip -- %f";
+    public static final String FILTER_LFS_PROCESS_KEY = "filter.lfs.process";
+    public static final String FILTER_LFS_PROCESS_VALUE = "git-lfs filter-process --skip";
+
+    private static final String LIST_GLOBAL_CONFIG_COMMAND = "git config --global --list";
+
+    public static final List<String[]> SKIP_SMUDGE_CONFIG_SETTINGS = Arrays.asList(
+            new String[] {FILTER_LFS_SMUDGE_KEY, FILTER_LFS_SMUDGE_VALUE},
+            new String[] {FILTER_LFS_PROCESS_KEY, FILTER_LFS_PROCESS_VALUE});
+
+    private static final Logger logger = LogsManager.getLogger(GitConfig.class);
+
+    /**
+     * Returns the configuration values for lfs smudge and process within global git config.
+     *
+     * @return a list of string arrays where 0-index is key and 1-index is value.
+     */
+    public static List<String[]> getGlobalGitLfsConfig() {
+        try {
+            String gitConfig = getGitGlobalConfig();
+            return Arrays.stream(gitConfig.split("\n"))
+                    .map(line -> line.split("="))
+                    .filter(line -> line[0].equals(FILTER_LFS_SMUDGE_KEY) || line[0].equals((FILTER_LFS_PROCESS_KEY)))
+                    .collect(Collectors.toList());
+        } catch (RuntimeException re) {
+            logger.log(Level.WARNING, "Could not get global git lfs config", re);
+            return new ArrayList<>();
+        }
+    }
+
+    /**
+     * Set the global git lfs configuration values. Does not run command if the configuration
+     * values are already set to skip.
+     *
+     * @param lfsConfigs a list of string arrays where 0-index is key and 1-index is value.
+     */
+    public static void setGlobalGitLfsConfig(List<String[]> lfsConfigs) {
+        String command = setGitLfsConfigCommand(lfsConfigs);
+        if (!command.equals("")) {
+            CommandRunner.runCommand(Paths.get("."), command);
+        }
+    }
+
+    /**
+     * Delete the global git lfs configuration values.
+     */
+    public static void deleteGlobalGitLfsConfig() {
+        String command = "git config --global --unset " + FILTER_LFS_SMUDGE_KEY
+                + " && git config --global --unset " + FILTER_LFS_PROCESS_KEY;
+        CommandRunner.runCommand(Paths.get("."), command);
+    }
+
+    private static String setGitLfsConfigCommand(List<String[]> lfsConfigs) {
+        String command = "";
+        for (String[] config : lfsConfigs) {
+            command += "git config --global " + config[0] + " " + "\"" + config[1] + "\"" + " && ";
+        }
+        return command.substring(0, Math.max(0, command.length() - 4));
+    }
```

Comment:
I'm not really a fan of how this function is written. A couple of points:
1. The string concatenation should be replaced with `String.format()` so that it is easier to read. Here's an example of how it's done elsewhere in the code:
https://github.com/reposense/RepoSense/blob/37425888df381680c7118a095ad3e5c5b0793cff/src/main/java/reposense/git/GitDiff.java#L32-L33
2. The command chaining is done in a rather hacky way. Can I suggest instead to append each command obtained from each iteration of `lfsConfigs` into an ArrayList, then use `String.join()` to chain the various commands together. We can then omit the use of `Math.max()` and `command.substring()` here.
3. Are we able to guarantee that the size of `lfsConfigs` would always be 2? What happens if the user has a malformed Git config where only the key exists? For example, you are expecting something like `filter.lfs.smudge=blah`, but what happens if the file only has `filter.lfs.smudge` without any `=`? I would suggest having an additional check here to ensure that we truly have a size of 2, otherwise we would get an IndexOutOfBoundsException.

For the last point, an example file could be something like this (open using `git config --global -e`):
```
[filter "lfs"]
        smudge
```
The output would just be `filter.lfs.smudge` on its own line.